File: FFmpegSdk/src/main/java/com/zh/ffmpegsdk/VideoPlayer.java
package com.zh.ffmpegsdk;

import android.view.Surface;

public class VideoPlayer {
    private VideoFrame.OnFrameDataCallBackListener onFrameDataCallBackListener;

    public String getInfo() {
        return getVersion();
    }

    public void startPlay(String path, Surface surface, VideoFrame.OnFrameDataCallBackListener onFrameDataCallBackListener) {
        this.onFrameDataCallBackListener = onFrameDataCallBackListener;
        start(path, surface);
    }

    private void frameDataCallBack(byte[] data, int[] linesize, int width, int height) {
        if (onFrameDataCallBackListener != null) {
            onFrameDataCallBackListener.call(data, linesize, width, height);
        }
    }

    public void stopPlay() {
        stop();
    }

    private native String getVersion();

    private native void start(String path, Surface surface);

    private native void stop();

    static {
        System.loadLibrary("media_handler");
    }
}

File: app/src/main/java/com/zhanghao/player/VideoPlayer.kt
package com.zhanghao.player

import android.content.Context
import android.util.AttributeSet
import android.view.Surface
import android.view.SurfaceView
import kotlin.concurrent.thread

class VideoPlayer(context: Context, attributeSet: AttributeSet) :
    SurfaceView(context, attributeSet) {
    fun startPlay(string: String) {
        thread {
            start(string, holder.surface)
        }
    }

    fun stopPlay() {
        stop()
    }

    fun getFFmpegVersion(): String {
        return getVersion()
    }

    private external fun getVersion(): String
    private external fun start(string: String, surface: Surface)
    private external fun stop()

    companion object {
        init {
            System.loadLibrary("media_player")
        }
    }
}